Pipeline Orchestration task

gridcell opened the pull request.

gridcell committed:
task setup

gridcell committed:
orchestration task

gridcell committed:
example pipeline json files

gridcell commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Orchestration Task](https://trello.com/c/wgnZMbVk/340-orchestration-task)

saanobhaai commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+FROM python:3.9-slim-bullseye
```
Any reason not to pull from `scl3/task_base` (https://github.com/SpeciesConservationLandscapes/task_base/blob/master/Dockerfile)?

saanobhaai commented on src/task.py:
```diff
@@ -0,0 +1,120 @@
+        super().__init__(*args, **kwargs)
+
+        self.pipeline = self._get_pipeline(kwargs)
+        self.exit_on_error = kwargs.get("exit_on_error") or False
```
Refactor to use base `raiseonfail` property?
https://github.com/SpeciesConservationLandscapes/task_base/blob/master/task_base/task.py#L36

saanobhaai commented on src/task.py:
```diff
@@ -0,0 +1,120 @@
+    def calc(self):
+        for task_group in self.pipeline:
+            if self.run_task_group(task_group) is False:
+                raise PipelineError("Task error")
```
I might need some reminding on where/why we're raising exceptions...

saanobhaai commented on src/task.py:
```diff
@@ -0,0 +1,120 @@
+                logging.error(err)
+                has_errors = True
+
+        return not has_errors or not self.exit_on_error
```
Would this be more readable as
`return has_errors and self.exit_on_error`
?

gridcell replied:
I want the `run_group` outcome to be False, showing that it successfully ran, `return has_errors and self.exit_on_error` would say run_group is True if it did not run successfully

saanobhaai commented on Dockerfile:
```diff
@@ -0,0 +1,12 @@
+RUN mkdir -p /app
+WORKDIR /app
+COPY ./src/* .
+COPY ./vm/scl-orchestration-vm.jinja .
```
I've managed to get confused... These files don't exist, but are part of the pipeline-devops. Maybe the Dockerfile and the devops/jinja stuff got moved but not cleaned up here?

saanobhaai reviewed (changes requested):
Awesome -- but a couple of questions

saanobhaai commented on Makefile:
```diff
@@ -0,0 +1,19 @@
+VM_IMAGE=scl/orchestration-vm
```
We should use `scl3` for consistency

gridcell committed:
pass env variables to tasks

gridcell committed:
set env vars as a list

saanobhaai committed:
change docker logging to syslog

gridcell committed:
remove unused files and add a few notes to the readme on usage

gridcell committed:
exit_on_error to raiseonfail, switched to log generator

gridcell committed:
tweaks

- update makefile to use --raiseonfail in local tests
- print out future results as they come in
